shared_model/interfaces/iroha_internal/transaction_batch.cpp

```diff
@@ -0,0 +1,136 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "interfaces/iroha_internal/transaction_batch.hpp"
+#include "validators/field_validator.hpp"
+#include "validators/transaction_validator.hpp"
+#include "validators/transactions_collection/batch_order_validator.hpp"
+
+namespace shared_model {
+  namespace interface {
+
+    /**
+     * check if all transactions belong to the same batch
+     * @param txs transactions to be checked
+     * @return true if all transactions from the same batch and false otherwise
+     */
+    static bool allTxsInSameBatch(const types::SharedTxsCollectionType &txs) {
+      // if txs is empty then true
+      if (txs.empty()) {
+        return true;
+      }
+
+      // take batch meta of the first transaction and compare it with batch
+      // metas of remaining transactions
+      auto batch_meta = txs.front()->batchMeta();
+      if (not batch_meta) {
+        return false;
+      }
+
+      for (auto &it = ++txs.begin(); it < txs.end(); it++) {
```

Comment:
This for may be reworked with std::accumulate. Also, the previous check, in this case, will be redundant.

Reply:
Can you be more precise please and explain algorithm? If your algorithm starts with `std::accumulate(txs.begin(), txs.end(),...)` then it is less efficient, because accumulate iterates over entire collection whereas it should be stopped when invalid tx is found